src/Factory/Consignment/ConsignmentFactory.php

```diff
@@ -250,7 +250,7 @@ private function getDeliveryType(): int
     {
         $deliveryTypeId = Arr::get(AbstractConsignment::DELIVERY_TYPES_NAMES_IDS_MAP, $this->deliveryOptions->getDeliveryType());
 
-        return $deliveryTypeId ?? AbstractConsignment::DELIVERY_TYPE_STANDARD;
+        return is_int($deliveryTypeId) ? $deliveryTypeId : AbstractConsignment::DELIVERY_TYPE_STANDARD;
```

Comment:
Dit is best wel raar want er komen alleen maar ints uit `AbstractConsignment::DELIVERY_TYPES_NAMES_IDS_MAP`, nu check je eigenlijk of hij iets heeft kunnen vinden of niet. Doe dan een check of `getDeliveryType` wel gevuld is.

Reply:
Arr::get is een gevaarlijke functie: als ie iets niet vindt, returnt ie de hele array, leek me beter om dat daarna op te pikken, niet vantevoren? heb er ook aan gedacht op getdeliverytype te checken, maar dan blijf je zitten met dat je daarna alsnog moet checken wat er uit Arr::get gekomen is, en dan kan je m.i. net zo goed niet op voorhand al gaan zitten checken.

Reply:
Want voor je het weet is ```getDeliveryType weldegelijk gevuld, maar vind ie toch niets... en dan hebben we weer dezelfde bug... Toch?

Reply:
`Arr::get` returnt standaard gewoon `null` als hij iets niet kan vinden toch? Hij returnt de hele array wel als `$key` `null` is. Dat is wel een heeeel rare keuze, we moeten een keer die helpers updaten (aka weggooien en composeren)...

Anyway, `getdeliveryType` is altijd `string | null`, dus als je dit doet werkt het toch?
```php
$deliveryType = $this->deliveryOptions->getDeliveryType();

if (! $deliveryType) {
    return null;
}

return Arr::get(AbstractConsignment::DELIVERY_TYPES_NAMES_IDS_MAP, $deliveryType);
```

Reply:
Dat is idd het alternatief. Maar omdat die Arr::get strict genomen niet under my control is, en idd een heel rare keuze heeft gemaakt (en wie weet wat nog meer later) had ik gekozen voor mijn optie. Maar nu werkt deze die je suggereert ook, dus ik pas t aan!